panamaconverter/src/PanamaCIRCTConverter.scala

```diff
@@ -123,7 +123,7 @@ class FirContext {
 
 class PanamaCIRCTConverter(val circt: PanamaCIRCT, fos: Option[FirtoolOptions], annotationsJSON: String) {
   val firCtx = new FirContext
-  val mlirRootModule = circt.mlirModuleCreateEmpty(circt.unkLoc)
+  var mlirRootModule = circt.mlirModuleCreateEmpty(circt.unkLoc)
```

Comment:
I don’t like this… would it still be `val`?

Reply:
It needs to be refactored later. `PanamaCIRCTConverter.newWithMlir` replaces this value. The currently code structure is indeed a bit dirty, I will refactor the whole structure of binder in the follow up PRs.